src/mango/src/mango_idx_view.erl

```diff
@@ -417,7 +417,7 @@ range(_, _, LCmp, Low, HCmp, High) ->
 % beginsWith requires both a high and low bound
 range({[{<<"$beginsWith">>, Arg}]}, LCmp, Low, HCmp, High) ->
     {LCmp0, Low0, HCmp0, High0} = range({[{<<"$gte">>, Arg}]}, LCmp, Low, HCmp, High),
-    range({[{<<"$lte">>, <<Arg/binary, 16#10FFFF>>}]}, LCmp0, Low0, HCmp0, High0);
+    range({[{<<"$lte">>, <<Arg/binary, 16#FFFF/utf8>>}]}, LCmp0, Low0, HCmp0, High0);
```

Comment:
That is not enough.  In addition to this, you must make sure that `Arg` is a well-formed UTF8 literal otherwise the combination will crash.  Also, it seems `16#10FFFF` is a valid Unicode value although it does not have a symbol assigned.  But that is the same for `16#FFFF`.

Reply:
For more details, see [how Erlang handles Unicode strings](https://www.erlang.org/doc/reference_manual/expressions#unicode-segments).

Reply:
yes - it's all a bit tricky ensuring that we don't double encode. I'll look at adding some tests for `range` directly.